File: packages/web/src/libs/types/message.ts
export type Message = {
  id: string;
  app: 'discord' | 'slack' | 'github';
  sender_image: string;
  sender_name: string;
  server_image: string;
  server_name: string;
  channel_name: string;
  content: string;
  message_link: string;
  sentiment: string;
  priority: 1 | 2 | 3 | 4 | 5;
  send_at: string;
};

export type MessageResponse = Omit<
  Message,
  'send_at' | 'message_link' | 'priority' | 'sentiment' | 'app'
> & {
  app: 'discord' | 'slack';
};

File: packages/api/src/const/message.py
from enum import Enum
from pydantic import BaseModel

class App(Enum):
    SLACK = "slack"
    DISCORD = "discord"


class Message(BaseModel):
    app: App
    server_id: str
    channel_id: str
    message_id: str


class MessageResponse(BaseModel):
    id: str
    app: App
    server_name: str
    server_image: str
    channel_name: str
    sender_name: str
    sender_image: str
    content: str
    message_link: str
    send_at: str